datus-clickhouse/tests/test_connector.py

```
@@ -0,0 +1,409 @@
+        ) ENGINE = MergeTree()
+        ORDER BY id
+    """
+    )
+
+    try:
+        schema = connector.get_schema(database_name=config.database, table_name=table_name)
+
+        assert len(schema) == 4
+
+        # Check id column
+        id_col = [col for col in schema if col["name"] == "flag"][0]
```

Ruff (0.14.8)

Prefer `next(col for col in schema if col["name"] == "flag")` over single element slice

Replace with `next(col for col in schema if col["name"] == "flag")`

(RUF015)